src/shogun/machine/IterativeLinearMachine.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Shubham Shukla
+ */
+
+#ifndef _ITERATIVELINEARMACHINE_H__
+#define _ITERATIVELINEARMACHINE_H__
+
+#include <shogun/lib/config.h>
+
+#include <shogun/lib/common.h>
+#include <shogun/machine/LinearMachine.h>
+#include <shogun/lib/SGVector.h>
+
+
+namespace shogun
+{
+  class CFeatures;
+  class CLabels;
+
+  class CIterativeLinearMachine : public CLinearMachine
+   {
+      public:
+        
+      /* default constructor */  
+       CIterativeLinearMachine();
+       virtual ~CIterativeLinearMachine();
+       
+       /** Train machine
+        * @params f features
+        * @params l labels 
+        * @return whether training was successful
+        */
+       virtual bool train(CDotFeatures* f, CLabels* l);
+       
+       void register_params();
+       
+       /* Continue Training
```

Comment:
This needs more documentation about when it fails and what it is expected to do. Furthermore, as discussed, it should be in CMachine, virtual, and throwing an error (SG_NOTIMPLEMENTED)